### PR TITLE
py-cheap-repr: new package

### PR DESCRIPTION
**Pull request by jmlapre**
Add [`cheap_repr`](https://pypi.org/project/cheap-repr/) Python library.  Required by [`snoop`](https://pypi.org/project/snoop/).  PR for snoop incoming soon.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-cheap-repr/package.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class PyCheapRepr(PythonPackage):
+    """This library provides short, fast, configurable string representations,
+    and an easy API for registering your own. It's an improvement of the
+    standard library module reprlib (repr in Python 2)."""
+
+    homepage = "https://pypi.org/project/cheap-repr/"
+    url = "https://pypi.org/packages/source/c/cheap-repr/cheap_repr-0.5.1.tar.gz"
+    list_url = "https://pypi.org/simple/cheap-repr/"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cheap-repr/package.py]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#id2

File: var/spack/repos/builtin/packages/py-cheap-repr/package.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    license("MIT", checked_by="jmlapre")
+
+    version("0.5.1", sha256="31ec63b9d8394aa23d746c8376c8307f75f9fca0b983566b8bcf13cc661fe6dd")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools-scm", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cheap-repr/package.py]:
```suggestion
    depends_on("py-setuptools-scm+toml", type="build")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by jmlapre — COMMENTED** [review]
Should be all fixed.

**Comment by jmlapre** [comment]
@adamjstewart Can you take a look at this PR when you get a moment?  Thanks.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**